geopandas/plotting.py

```diff
@@ -109,7 +125,6 @@ def plot_polygon_collection(
             "'pip install descartes'."
```

Comment:
A question about the signatures, e.g. of `plot_polygon_collection`:

* Is there a reason `color` is listed as a seperate parameter, but all other style elements are assumed in `kwargs`?
* I find this especially puzzling as I noticed that none of the callers of this function, *at least from within the module*, pass a `color` value, and all other parameters (after `color`) are passed by keyword.
* That would make it possible to remove the `color` from the signature, and treat it like any other style parameter.
* Because the function name is not prefixed with a `_`, however, I think I should assume there are possibly other callers that imported the module.

Any thoughts on this? Especially the last point.

Reply:
> Is there a reason color is listed as a seperate parameter, but all other style elements are assumed in kwargs?

Can't see a reason for that. Might be for historical reasons.

> I find this especially puzzling as I noticed that none of the callers of this function, at least from within the module, pass a color value, and all other parameters (after color) are passed by keyword.

It has to be part of kwargs for `PatchCollection`, so while we allow it to be set by defined parameter, we have to put it to kwargs.

> That would make it possible to remove the color from the signature, and treat it like any other style parameter.

I guess we can do that in future. We have made these private only recently (#656), so I don't wan to change the API now.

> Because the function name is not prefixed with a _, however, I think I should assume there are possibly other callers that imported the module.

It is prefixed on master. The change has been made in #656 and merged few days ago. I have updated this branch to accommodate those changes.

There have been a lot of piecemeal changes to plotting, not all of them actually helped in terms of code clarity and consistency.